pkg/client/logs.go

```diff
@@ -84,20 +83,43 @@ func (r *Reader) Read(p []byte) (int, error) {
 		return n, nil
 	}
 
-	data, ok := <-r.bytestream
-	// If the bytestream is done then save the error for future calls to Read.
-	if !ok {
-		r.err, ok = <-r.errc
-		// Assume EOF if error channel is closed.
-		if r.err == nil && !ok {
+	// Return the error if set only if byte channel is empty.
+	if r.err != nil && r.empty {
+		return 0, r.err
+	}
+
+	// Be sure that we are draining any errors put on the errc and unblock senders.
+	var err error
+	var data []byte
+	var ok bool
+	select {
+	case err, ok = <-r.errc:
+		// Treat the errors as data as well. This ensures that the user reading the messages
+		// is aware of the errors as they happen and not only after all messages are processed.
+		// This is important in the case of tailing logs where one stream may never finish at all.
```

Comment:
This is the contentious part of this PR; printing these as we get them. I think it is the most logical thing to do with the errors since we may be tailing 10+ streams. Even in the default case on a 2 node cluster we have the aggregator, e2e tests + sonobuoy sidecar, 2 containers to get logs and their 2 sidecars. So 7 is a minimum.

If we just hold all errors until the end; 6/7 could be failing for some meaningful reason but the user would see a blank screen just seeing the logs of the 7th container doing (maybe) nothing.